src/crewai/tasks/task_guardrail.py

```diff
@@ -0,0 +1,102 @@
+from typing import Any, Optional, Tuple
+
+from pydantic import BaseModel, Field
+
+from crewai.agent import Agent, LiteAgentOutput
+from crewai.llm import LLM
+from crewai.task import Task
+from crewai.tasks.task_output import TaskOutput
+
+
+class TaskGuardrailResult(BaseModel):
+    valid: bool = Field(
+        description="Whether the task output complies with the guardrail"
+    )
+    feedback: str | None = Field(
+        description="A feedback about the task output if it is not valid",
+        default=None,
+    )
+
+
+class TaskGuardrail:
+    """A task that validates the output of another task using generated Python code.
+
+    This class is used to validate the output from a Task based on specified criteria.
+    It uses an LLM to validate the output and provides a feedback if the output is not valid.
+
+    Args:
+        description (str): The description of the validation criteria.
+        task (Task, optional): The task whose output needs validation.
+        llm (LLM, optional): The language model to use for code generation.
+    """
+
+    def __init__(
+        self,
+        description: str,
+        task: Task | None = None,
```

Comment:
I see now. If you don't actually need the task for anything, let's move this logic up.

Reply:
I'm using it as mentioned [here ](https://github.com/crewAIInc/crewAI/pull/2636#discussion_r2066997649)